packages/react/src/components/Tree/TreeItem.tsx

```diff
@@ -0,0 +1,250 @@
+   */
+  renderItemTitle?: ShorthandRenderFunction
+
+  /** Properties for TreeTitle. */
+  title?: ShorthandValue<TreeTitleProps>
+}
+
+class TreeItem extends UIComponent<WithAsProp<TreeItemProps>> {
+  static create: Function
+
+  static displayName = 'TreeItem'
+
+  static className = 'ui-tree__item'
+
+  static slotClassNames: TreeItemSlotClassNames = {
+    title: `${TreeItem.className}__title`,
+    subtree: `${TreeItem.className}__subtree`,
+  }
+
+  static propTypes = {
+    ...commonPropTypes.createCommon({
+      content: false,
+    }),
+    items: customPropTypes.collectionShorthand,
+    indexInTree: PropTypes.number,
+    exclusive: PropTypes.bool,
+    level: PropTypes.number,
+    onTitleClick: PropTypes.func,
+    onFirstChildFocus: PropTypes.func,
+    onParentFocus: PropTypes.func,
+    onSiblingsExpand: PropTypes.func,
+    open: PropTypes.bool,
+    parent: customPropTypes.itemShorthand,
+    indexInSubtree: PropTypes.number,
+    renderItemTitle: PropTypes.func,
+    siblings: customPropTypes.collectionShorthand,
+    treeItemRtlAttributes: PropTypes.func,
+    title: customPropTypes.itemShorthand,
+  }
+
+  static defaultProps = {
+    as: 'div',
+    accessibility: treeItemBehavior,
+  }
+
+  itemRef = React.createRef<HTMLElement>()
+  treeRef = React.createRef<HTMLElement>()
+
+  actionHandlers = {
+    performClick: e => {
+      e.preventDefault()
+      e.stopPropagation()
+
+      this.handleTitleClick(e)
+    },
+    focusParent: e => {
+      e.preventDefault()
+      e.stopPropagation()
+
+      this.handleParentFocus(e)
+    },
+    collapse: e => {
+      e.preventDefault()
+      e.stopPropagation()
+
+      this.handleTitleClick(e)
+    },
+    expand: e => {
+      e.preventDefault()
+      e.stopPropagation()
+
+      this.handleTitleClick(e)
+    },
+    focusSubtree: e => {
+      e.preventDefault()
+      e.stopPropagation()
+
+      this.handleFirstChildFocus(e)
+    },
+    expandSiblings: e => {
+      e.preventDefault()
+      e.stopPropagation()
+
+      this.handleSiblingsExpand(e)
+    },
+  }
+
+  eventComesFromChildItem = e => {
+    return e.currentTarget !== e.target
+  }
+
+  handleTitleClick = e => {
+    _.invoke(this.props, 'onTitleClick', e, this.props)
+  }
+
+  handleParentFocus = e => {
+    _.invoke(this.props, 'onParentFocus', e, this.props)
+  }
+
+  handleFirstChildFocus = e => {
+    _.invoke(this.props, 'onFirstChildFocus', e, this.props)
+  }
+
+  handleSiblingsExpand = e => {
+    _.invoke(this.props, 'onSiblingsExpand', e, this.props)
+  }
+
+  handleTitleOverrides = (predefinedProps: TreeTitleProps) => ({
+    onClick: (e, titleProps) => {
+      this.handleTitleClick(e)
+      _.invoke(predefinedProps, 'onClick', e, titleProps)
+    },
+  })
+
+  renderContent() {
+    const {
+      items,
+      title,
+      renderItemTitle,
+      open,
+      level,
+      siblings,
+      indexInTree,
+      indexInSubtree,
+    } = this.props
+    const hasSubtree = !_.isNil(items)
```

Comment:
What if the collection is empty?

Reply:
it was like this in HierarchicalTree but sure I will add an extra check thanks!